tests/dataset_builders/pie/test_cdcp.py

```diff
@@ -0,0 +1,418 @@
+from pie_datasets.document.types import TokenDocumentWithLabeledSpansAndBinaryRelations
+from tests import FIXTURES_ROOT
+from tests.dataset_builders.common import PIE_BASE_PATH, _deep_compare
+
+disable_caching()
+
+DATASET_NAME = "cdcp"
+SPLIT_SIZES = {"train": 581, "test": 150}
+HF_DATASET_PATH = CDCP.BASE_DATASET_PATH
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME  # "pie/cdcp"
+DATA_PATH = FIXTURES_ROOT / "dataset_builders" / "cdcp_acl17.zip"
+
+HF_EXAMPLE_00195 = {
+    "id": "00195",
+    "text": "State and local court rules sometimes make default judgments much more likely. For example, "
+    "when a person who allegedly owes a debt is told to come to court on a work day, they may be "
+    "forced to choose between a default judgment and their job. I urge the CFPB to find practices "
+    "that involve scheduling hearings at inconvenient times unfair, deceptive, and abusive, or "
+    "inconsistent with 1692i.",
+    "propositions": {
+        "start": [0, 78, 242],
+        "end": [78, 242, 391],
+        "label": [4, 4, 1],
+        "url": ["", "", ""],
+    },
+    "relations": {"head": [0, 2], "tail": [1, 0], "label": [1, 1]},
+}
+
+
+HF_EXAMPLE_00194 = {
+    "id": "00194",
+    "text": "Recently, courts have held that debt collectors can escape 1692i's venue provisions entirely "
+    "by pursuing debt collection through arbitration instead. As the NAF studies reflect, arbitration "
+    "has not proven a satisfactory alternative. I urge the CFPB to include in a rule language "
+    "interpreting 1692i as requiring debt collectors to proceed in court, not through "
+    "largely-unregulated arbitral forums.",
+    "propositions": {
+        "start": [0, 149, 232],
+        "end": [149, 232, 396],
+        "label": [0, 4, 1],
+        "url": ["", "", ""],
+    },
+    "relations": {"head": [2], "tail": [1], "label": [1]},
+}
+
+
+@pytest.fixture(scope="module", params=["train", "test"])
+def split(request):
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def hf_dataset():
+    return load_dataset(str(HF_DATASET_PATH), data_dir=DATA_PATH)
+
+
+def test_hf_dataset(hf_dataset):
+    assert hf_dataset is not None
+    assert {name: len(ds) for name, ds in hf_dataset.items()} == SPLIT_SIZES
+
+
+@pytest.fixture(scope="module")
+def hf_example(hf_dataset, split):
+    return hf_dataset[split][0]
+
+
+def test_hf_example(hf_example, split):
+    assert hf_example is not None
+    if split == "train":
+        assert hf_example == HF_EXAMPLE_00195
+    elif split == "test":
+        assert hf_example == HF_EXAMPLE_00194
+    else:
+        raise ValueError(f"Unknown split: {split}")
+
+
+@pytest.fixture(scope="module")
+def generate_document_kwargs(hf_dataset, split):
+    return CDCP()._generate_document_kwargs(hf_dataset[split])
+
+
+def test_example_to_document(hf_example, generate_document_kwargs):
+    doc = example_to_document(hf_example, **generate_document_kwargs)
+    assert doc is not None

```

Comment:
Please align this with the way it is done here: https://github.com/ArneBinder/pie-datasets/blob/main/tests/dataset_builders/pie/test_conll2003.py#L51-L69.

EDIT: I just noticed, that we already have the `document` fixture below. So, maybe name one fixture `document` and the other `generated_document` and then also have both methods, `test_document` and `test_generated_document`, but just in one test the actual content and in the other just compare to the otehr fixture content.

Reply:
added in 083b80f